cheroot/server.py

```diff
@@ -1810,6 +1811,14 @@ def serve(self):
                     traceback=True,
                 )
 
+            # raise exceptions reported by any worker threads,
+            # such that the exception is raised from the serve() thread.
+            if self.interrupt:
+                while self._stopping_for_interrupt:
+                    time.sleep(0.1)
+                if self.interrupt:
+                    raise self.interrupt
```

Comment:
Let's avoid repeating the recent regression:
```suggestion
                    self.serving = False
                    raise self.interrupt
```